src/Hazelcast.Net/Protocol/CustomCodecs/SqlErrorCodec.cs

```diff
@@ -66,16 +67,16 @@ public static Hazelcast.Sql.SqlError Decode(IEnumerator<Frame> iterator)
 
             var originatingMemberId = initialFrame.Bytes.ReadGuidL(OriginatingMemberIdFieldOffset);
             var message = CodecUtil.DecodeNullable(iterator, StringCodec.Decode);
-            string suggestion = String.Empty;
-
-            if (!iterator.AtStructEnd())
+            var isSuggestionExists = false;
+            string suggestion = default;
+            if (iterator.NextIsNotTheEnd())
```

Comment:
I think this shifts the frames more than needed. `StringCodec.Decode` uses `Frame.Take` which takes current frame and shifts to next. When combined with  `NextIsNotTheEnd+Decode` makes two shift. OTOH, `Frame.AtStructEnd` just peeks to current frame.

Reply:
`NextIsNotTheEnd` had a bug, weird enough, that I fixed, and now this passes all tests. Or can you think of a test that we could get to fail?

Reply:
I think that is enough as long as that is not shift the position.